src/PostProcesses/postProcess.ts

```diff
@@ -677,17 +732,32 @@ export class PostProcess {
 
     private _disposeTextures() {
         if (this._shareOutputWithPostProcess || this._forcedOutputTexture) {
+            this._disposeTextureCache();
             return;
         }
 
         if (this._textures.length > 0) {
             for (var i = 0; i < this._textures.length; i++) {
                 this._engine._releaseTexture(this._textures.data[i]);
+
+                const cacheIndex = this._textureCache.indexOf(this._textures.data[i]);
+                if (cacheIndex !== -1) {
+                    this._textureCache.splice(cacheIndex, 1);
+                }
             }
         }
+
+        this._disposeTextureCache();

```

Comment:
Is the code inside the loop above necessary given that `this._disposeTextureCache();` is called unconditionally here?

Reply:
yeah, `_textureCache` should always contain `_textures`, I will remove the loop